### PR TITLE
changes for production mode

### LINKED ISSUE #69: Run the application in production mode
It won't be long before we will want to run the application in production mode.  We need the following changes (see the App Portfolio or AAEC for examples of how these changes were made):

* ~~Add the mysql2 gem~~
* tweak the config/puma.rb
* edit config.ru to use RAILS_RELATIVE_URL_ROOT
* tweak settings in config/environments/production.rb
* Add robots.txt
* create tmp/puma/.keep

Resources
* App Portfolio: https://github.com/uclibs/application_portfolio/pull/101 & https://github.com/uclibs/application_portfolio/pull/124 & https://github.com/uclibs/application_portfolio/pull/159
* AAEC: https://github.com/uclibs/aaec/pull/70 & https://github.com/uclibs/aaec/pull/85

### PR DESCRIPTION
**Pull request by crowesn**
Fixes #69 

Add config changes to run application in production mode.

Changes proposed in this pull request:
* config production values
* Add dotenv variables for production mode
* Add readme entry about docker production mode

### CODE REVIEW

**Comment by crowesn** [comment]
Probably best to test manually on local machine

Production mode: `RAILS_ENV=production docker-compose up -d --build`
Development mode: `docker-compose up -d --build`

I don't think you should need to clear the data between tests but you can stop the app with `docker-compose down` to retain data and `docker-compose down -v` to delete the data.

File: config/puma.rb [hunk 1]
```diff
@@ -8,17 +8,28 @@
...
-port        ENV.fetch('PORT', 3000)
+if ENV.fetch('RAILS_ENV') == 'production'
+  # Set up socket location
+  bind "unix://#{app_dir}/tmp/puma/puma.sock"
+
+  # Logs
+  stdout_redirect "#{app_dir}/log/puma.stdout.log", "#{app_dir}/log/puma.stderr.log", true
+else
+  # Specifies the `port` that Puma will listen on to receive requests; default is 3000.
+  port ENV.fetch('PORT', 3010)
```
Comment (hortongn) [inline config/puma.rb:23]: I think this should be 3000 not 3010.

**Review by hortongn — CHANGES_REQUESTED** [review]
@crowesn I had to make a couple tweaks to the puma.rb to get the site to load for me.  See my comments inline.  We can discuss if those suggestions don't make sense.

**Review by hortongn — APPROVED** [review]
@crowesn Works great.  Nice job!

**Pull request merged**